catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -612,14 +589,30 @@ private ResourceInfo getResourceInfo(
           "Could not resolve source id for URI by doing a URI based query.");
     }
     metacard = queryResponse.getResults().get(0).getMetacard();
-    if (uri != null && queryResponse.getResults().size() > 1) {
-      for (Result result : queryResponse.getResults()) {
-        if (uri.equals(result.getMetacard().getResourceURI())) {
-          metacard = result.getMetacard();
-          break;
+    if (queryResponse.getResults().size() > 1) {
+      if (uri != null) {
+        for (Result result : queryResponse.getResults()) {
+          if (uri.equals(result.getMetacard().getResourceURI())) {
+            metacard = result.getMetacard();
+            break;
+          }
         }
+      } else {
+        throw new UnsupportedQueryException(
+            "A resource request resulted in multiple metacards being returned.");
```

Comment:
This else doesn't look to me like it's lined up in the right place. Should this be 
```java
    if (queryResponse.getResults().size() > 1) {
      if (uri != null) {
        for (Result result : queryResponse.getResults()) {
          if (uri.equals(result.getMetacard().getResourceURI())) {
            metacard = result.getMetacard();
            break;
          }
        }
      }
    } else {
      throw new UnsupportedQueryException(
              "A resource request resulted in multiple metacards being returned.");
    }
```
 instead?